pkg/apis/core/validation/validation.go

```diff
@@ -2687,38 +2687,48 @@ func validateAffinityTimeout(timeout *int32, fldPath *field.Path) field.ErrorLis
 	return allErrs
 }
 
-// AccumulateUniqueHostPorts extracts each HostPort of each Container,
-// accumulating the results and returning an error if any ports conflict.
-func AccumulateUniqueHostPorts(containers []core.Container, accumulator *sets.String, fldPath *field.Path) field.ErrorList {
+// checkPortConflicts looks for duplicate hostPorts and duplicate namedPorts.
+// It returns an error list of conflicting port definitions.
+func checkPortConflicts(containers []core.Container, fldPath *field.Path) field.ErrorList {
```

Comment:
This change doesn't seem correct... previously, it was narrowly checking for host port conflicts. Now, the function has a generic name, but:

1. it isn't actually checking for unique containerPort+protocol conflicts (which is what the API declares to be the unique key for items in the containerPort list)
2. it is tightening validation on unique port names without maintaining compatibility with existing persisted data, and without making a clear case for this being a fatal error

I vaguely remember a discussion with @thockin (I thought it was in an issue, but I couldn't find it while searching) about what could break if ports shadowed earlier ports' names. Some ports that share a name being unaddressable seems like a reasonable way to resolve poorly specified ports, so I don't think I'd consider this a fatal error.

Warning when there are conflicting port names seems reasonable to me.